bin/i18n/utils/sync_up_base.rb

```diff
@@ -0,0 +1,92 @@
+require_relative '../i18n_script_utils'
+require_relative '../metrics'
+require_relative '../utils/crowdin_client'
+
+module I18n
+  module Utils
+    class SyncUpBase
+      Config = Struct.new :crowdin_project, :source_paths, :ignore_paths, keyword_init: true
+
+      MAIN_CROWDIN_PROJECT = 'codeorg'.freeze
+
+      def self.config
+        @config ||= Config.new(
+          crowdin_project: MAIN_CROWDIN_PROJECT,
+          source_paths: [],
+          ignore_paths: [],
+        )
+      end
+
+      def self.perform
+        i18n_resource_name = name[/^.*::(\w+::\w+)::SyncUp/, 1] || name
+
+        I18n::Metrics.report_runtime(i18n_resource_name, 'sync-up') do
+          new.send(:perform)
+        end
+      end
+
+      protected
+
+      def perform
+        progress_bar.start
+
+        crowdin_client # Initialize `@crowdin_client` outside threads to avoid multiple instance creations
+
+        I18nScriptUtils.process_in_threads(source_files, in_threads: PARALLEL_PROCESSES) do |source_file_path|
+          crowdin_client.upload_source_file(source_file_path)
+        ensure
+          mutex.synchronize {progress_bar.increment}
+        end
+
+        progress_bar.finish
+      end
+
+      private
+
+      PROGRESS_BAR_FORMAT = '%t: |%W| %c/%C %a'.freeze
+      PARALLEL_PROCESSES = 20 # https://developer.crowdin.com/api/v2/#section/Introduction/Rate-Limits
+
+      def config
+        self.class.config
+      end
+
+      def testing?
+        CDO.rack_env?(:development)
```

Comment:
`CDO.rack_env` is also `development` on the `i18n server` because we don't deploy it.
We need to find another solution for checking this, using the `command line arguments` for example.

Reply:
Are you planning to implement that solution as part of this PR?

Reply:
Yes, I replaced this check with the `AWS EC2 instance presence` check.
And we are going to add `command line arguments` for this in the following [task](https://codedotorg.atlassian.net/browse/P20-634).